packages/hoppscotch-common/src/services/banner.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Service } from "dioc"
+import { ref } from "vue"
+
+export type BannerType = "info" | "warning" | "error"
+
+export type BannerContent = {
+  type: BannerType
+  text: string
+  // Can be used to display an alternate text when display size is small
+  alternateText?: string
+}
+
+export class BannerService extends Service {
+  public static readonly ID = "BANNER_SERVICE"
+  constructor() {
+    super()
+  }
+
+  public content = ref<BannerContent | undefined>(undefined)
```

Comment:
`public` properties of services should have JSDoc comments describing what it does.